Base/QTGUI/qSlicerExportNodeDialog.cxx

```diff
@@ -733,6 +780,7 @@ bool qSlicerExportNodeDialogPrivate::populateNodeTypeWidgetSets()
     }
   else
     {
+    this->setWindowTitle(QString("Export 0 Nodes")); // This hopefully never happens, but we want a correct title in case it does.
     qCritical() << Q_FUNC_INFO << "failed: There is nothing to export.";
```

Comment:
If you are not sure if warning/error message should be used then a good test is to check the return code (if there is no return code then you make the decision based on what return code you would use):
- if you return with success (you can complete the request, but you suspect that something may be wrong or you encountered an error but you could recover from it) then it is more appropriate to use a warning log; it does not matter that you did not expect this request - if you can perform what was asked then it is not really an error
- if you return with failure (you cannot complete what the method promises to do) then an error message is more appropriate

Qt distinguishes a special error message type - qFatal - that should be logged if the application has to be terminated because of that error. So, the generic error message is qCritical. Since qCritical sounds very serious, sometimes developers use qWarning even for irrecoverable error occurs. Since there is no complete consistency in the Slicer code base; and log type here does not change the behavior of the application in any way (there are only a very few places where it actually changes the behavior), I'm OK with either choice.

Reply:
Thank you for these very helpful guidelines!

Then I think "error" is suitable in this case, so I will leave this part unchanged.